### PR TITLE
Fixed CLI login for Node.js 10 on Windows

### PR DESCRIPTION
**Pull request by alex-margaritov**
**Problem:** The CLI login process stalls when the user is on Node.js 10. In addition, CLI can't  be built via nodejs 10.

**Root-causes:** 
- NodeJS 10 introduced some breaking changes, e.g. `writeFile` requires callback now.
- Inquirer package 3.x.x is not compatible with nodeJS 10 on Windows machines.

**Changes:**
- Added explicit `natives` package instead of one from dependecies
- Replaced archived `rx-lite` with `rxjs@5.X`
- Replaced `writeFile` with `writeFileSync`
- Updated `inquirer`
- Updated typings
- Updated tests

### CODE REVIEW

**Comment by AndreyMitsyk** [comment]
Hey, @owenniblock ! Could you please review this PR? 😄

File: package-lock.json [hunk 1]
```diff
@@ -29,17 +37,26 @@
     },
     "@sinonjs/formatio": {
       "version": "2.0.0",
-      "resolved": "https://registry.npmjs.org/@sinonjs/formatio/-/formatio-2.0.0.tgz",
+      "resolved": "http://registry.npmjs.org/@sinonjs/formatio/-/formatio-2.0.0.tgz",
```
Comment (owenniblock) [inline package-lock.json]: @AndreyMitsyk - any idea why the paths in your `package-lock.json` are resolving as `http` instead of `https`?

Comment (AndreyMitsyk) [reply package-lock.json]: Don know, it was resolved automatically. @alex-margaritov do you have an idea why it happened?

Comment (AndreyMitsyk) [reply package-lock.json]: never mind. Regenerated file and now it looks fine.

Comment (owenniblock) [reply package-lock.json]: Great. Thanks @AndreyMitsyk - approving!

**Comment by owenniblock** [comment]
I'll merge once the build has completed :)

**Pull request merged**